### PR TITLE
Mas.workloads1

### PR DESCRIPTION
**Pull request by honcao**
The change I made is including:
1.	Create the Vms in parallel 
2.	Deploy applications in parallel
3.	Update the DSC extension version
4.	Update DSC LocalConfigurationManager setting.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@honcao__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by singhkay** [comment]
@kenazk @simongdavies Can you take a look at this as well before we merge?

**Comment by honcao** [comment]
Hi Kenaz and Simon

As we discussed, I submit the Pull request to update the SQL always on template.

Would you please help to review it and let me know if there is any issue I need to address?

I would like to merge the change to master branch ASAP.

Thanks
Hongbin

**Comment by azurecla** [comment]
__@honcao__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, AZPRBOT;

**Comment by simongdavies** [comment]
@singhkay @honcao  Looks good to me

File: sql-server-2014-alwayson-dsc/ADVmTemplate.json [hunk 1]
```diff
@@ -0,0 +1,367 @@
+﻿{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+
+    "VMNamePrefix": {
```
Comment (singhkay) [inline sql-server-2014-alwayson-dsc/ADVmTemplate.json]: Please camel case the parameters

File: sql-server-2014-alwayson-dsc/azuredeploy.json [hunk 2]
```diff
@@ -163,83 +196,46 @@
...
+        "East Asia",
+        "Southeast Asia",
+        "East US 2"
+      ],
       "metadata": {
-        "description": "The DNS Prefix for the Public IP Address"
-      }
+        "description": "The region to deploy the resources into"
+      },
+      "defaultValue": "East US 2"
```
Comment (singhkay) [inline sql-server-2014-alwayson-dsc/azuredeploy.json]: Please remove default value for location and let user choose it.

File: sql-server-2014-alwayson-dsc/azuredeploy.json [hunk 3]
```diff
@@ -320,607 +334,225 @@
...
-          "dnsServerAddress": {
-            "value": [
-              "[variables('adPDCNicIPAddress')]"
-            ]
-          }
+          "location": { "value": "[parameters('location')]" },
+          "adminUsername": { "value": "[parameters('adminUsername')]" },
+          "adminPassword": { "value": "[parameters('adminPassword')]" },
+          "storageAccount": { "value": "[variables('SqlWitnessStorageName')]" },
+          "SubnetUri": { "value": "[variables('sqlSubnetRef')]" },
```
Comment (singhkay) [inline sql-server-2014-alwayson-dsc/azuredeploy.json]: camel case for variables as well

File: sql-server-2014-alwayson-dsc/azuredeploy.parameters.json [hunk 4]
```diff
@@ -5,6 +5,15 @@
     "newStorageAccountNamePrefix": {
       "value": ""
     },
+
```
Comment (singhkay) [inline sql-server-2014-alwayson-dsc/azuredeploy.parameters.json]: Please fill out values for all parameters. For things that require a unique string such as storage accounts and dns names, please fill in "#####" (5 #) to enable automated testing

File: sql-server-2014-alwayson-dsc/azuredeploy.json [hunk 5]
```diff
@@ -150,12 +128,67 @@
...
     },
+
     "sqlServerServiceAccountPassword": {
       "type": "securestring",
       "metadata": {
         "description": "The SQL Server Service account password"
       }
     },
+
+    "VmNetworkSubnet": {
```
Comment (singhkay) [inline sql-server-2014-alwayson-dsc/azuredeploy.json]: Can this and the addresses below become a variable and not ask the user to enter so many parameters?

File: sql-server-2014-alwayson-dsc/azuredeploy.json [hunk 6]
```diff
@@ -163,83 +196,46 @@
       },
       "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/sql-server-2014-alwayson-dsc"
     },
-    "dnsPrefix": {
+
+    "location": {
       "type": "string",
```
Comment (singhkay) [inline sql-server-2014-alwayson-dsc/azuredeploy.json]: I think more regions are available now for CRP, so we can add to this list

File: sql-server-2014-alwayson-dsc/ADVmTemplate.json (hunk 1 above)
Comment (honcao) [inline sql-server-2014-alwayson-dsc/ADVmTemplate.json]: Fixed.

File: sql-server-2014-alwayson-dsc/azuredeploy.json (hunk 2 above)
Comment (honcao) [inline sql-server-2014-alwayson-dsc/azuredeploy.json]: Fxied

File: sql-server-2014-alwayson-dsc/azuredeploy.json (hunk 3 above)
Comment (honcao) [inline sql-server-2014-alwayson-dsc/azuredeploy.json]: Fxied

File: sql-server-2014-alwayson-dsc/azuredeploy.parameters.json (hunk 4 above)
Comment (honcao) [inline sql-server-2014-alwayson-dsc/azuredeploy.parameters.json]: Fxied

File: sql-server-2014-alwayson-dsc/azuredeploy.json (hunk 6 above)
Comment (honcao) [inline sql-server-2014-alwayson-dsc/azuredeploy.json]: Fxied

File: sql-server-2014-alwayson-dsc/azuredeploy.json (hunk 5 above)
Comment (honcao) [inline sql-server-2014-alwayson-dsc/azuredeploy.json]: I will not fix this comment. I would like to keep those as parameters and the values can be changed during deployment

**Pull request merged**